Backend/WebApi.Weather/WeatherClient.cs

```diff
@@ -0,0 +1,142 @@
+        {
+            Latitude = latitude,
+            Longitude = longitude
+        };
+        return await QueryAsync(options);
+    }
+
+    public WeatherForecastApi? Query(WeatherForecastOptionsApi optionsApi)
+    {
+
+        return QueryAsync(optionsApi).GetAwaiter().GetResult();
+    }
+
+    public WeatherForecastApi? Query(float latitude, float longitude)
+    {
+        return QueryAsync(latitude, longitude).GetAwaiter().GetResult();
+    }
+
+
+    public string WeatherCodeToString(int weatherCode)
+    {
+        if (Enum.IsDefined(typeof(WeatherCode), weatherCode))
+        {
+            return ((WeatherCode)weatherCode).ToString().Replace('_', ' ');
+        }
+        else
+        {
+            return "Invalid weather code";
+        }
+    }
+    
+
+    public async Task<WeatherForecastApi?> GetWeatherForecastAsync(WeatherForecastOptionsApi optionsApi)
+    {
+        try
+        {
+            var response = await client.GetAsync(MergeUrlWithOptions(_weatherApiUrl, optionsApi));
+            response.EnsureSuccessStatusCode();
+
+            var weatherForecast = await JsonSerializer.DeserializeAsync<WeatherForecastApi>(
+                await response.Content.ReadAsStreamAsync(),
+                new JsonSerializerOptions { PropertyNameCaseInsensitive = true });
+            return weatherForecast;
+        }
+        catch (HttpRequestException e)
+        {
+            logger.LogError(e.Message);
+            logger.LogError(e.StackTrace);
+            return null;
+        }
+    }
+
+
+    private string MergeUrlWithOptions(string url, WeatherForecastOptionsApi? options)
+    {
+
+        if (options == null) return url;
+
+        var queryString = QueryHelpers.ParseQuery("?");
+        queryString.Add("latitude", options.Latitude.ToString(CultureInfo.InvariantCulture));
+        queryString.Add("longitude", options.Longitude.ToString(CultureInfo.InvariantCulture));
+        queryString.Add("temperature_unit", options.TemperatureUnit.ToString());
+        queryString.Add("precipitation_unit", options.PrecipitationUnit.ToString());
+        if (options.Timezone != string.Empty)
+            queryString.Add("timezone", options.Timezone);
+        queryString.Add("timeformat", options.Timeformat.ToString());
+        queryString.Add("past_days", options.PastDays.ToString());
+
+        if (options.StartDate != string.Empty)
+            queryString.Add("start_date", options.StartDate);
+        if (options.EndDate != string.Empty)
+            queryString.Add("end_date", options.EndDate);
+
+        // Now we iterate through minutely15
+
+        // minutely15
+        if (options.Minutely15.Count > 0)
+        {
+            var firstminutely15Element = true;
+            
+            String queryValue = null; 
+            foreach (var option in options.Minutely15)
+                if (firstminutely15Element)
+                {
+                    queryValue += option.ToString();
```

Comment:
https://learn.microsoft.com/en-us/dotnet/api/system.text.stringbuilder?view=net-8.0

Reply:
Zmieniłem ale nie wiem czy oto Ci chodziło

Reply:
nie o to, ale możesz olać, i tak robimy słabej jakości produkt